server/utils/APIWrapper.ts

```diff
@@ -1,33 +1,93 @@
+import Ajv, { ErrorObject, JTDSchemaType } from "ajv/dist/jtd";
 import mongoose from "mongoose";
 import { NextApiResponse } from "next";
 import { getSession } from "next-auth/client";
 import {
-  HttpMethod,
   InternalRequest,
   InternalResponseData,
   Role,
+  StringsDict,
+  UnknownDict,
 } from "src/utils/types";
 
-interface RouteConfig {
+interface RouteConfig<Q_P, B> {
   requireSession: boolean;
   roles?: Array<Role>;
   handleResponse?: boolean; // handleResponse if the route handles setting status code and body
+  bodySchema?: JTDSchemaType<B>;
+  querySchema?: JTDSchemaType<Q_P>;
 }
 
-interface Route<T> {
-  config?: RouteConfig;
+export interface Route<Q_P, B, R> {
+  config?: RouteConfig<Q_P, B>;
   handler: (
-    req: InternalRequest,
-    res: NextApiResponse<InternalResponseData<T>>
-  ) => Promise<T>;
+    req: InternalRequest<Q_P, B>,
+    res: NextApiResponse<InternalResponseData<R>>
+  ) => Promise<R>;
 }
 
-function APIWrapper(
-  routeHandlers: Partial<Record<HttpMethod, Route<unknown>>>
+export type UnvalidatedInputRoute = {
+  config: Omit<
+    RouteConfig<StringsDict, UnknownDict>,
+    "bodySchema" | "querySchema"
+  >;
+  handler: (
+    req: InternalRequest<StringsDict, UnknownDict>,
+    res: NextApiResponse
+  ) => Promise<unknown>;
+};
+
+/**
+ * @deprecated Please import { APIWrapper } instead, which enforces schemas
+ * Does not validate input schemas for the API
+ * @param routeHandlers
+ */
+export function UnvalidatedInputAPIWrapper(
+  routeHandlers: Partial<
+    Record<"GET" | "PATCH" | "POST" | "PUT", UnvalidatedInputRoute>
```

Comment:
Patch for the routes that don't have validation. TypeScript type inference does not play friendly with this (it infers `Route<any, any, any>` instead of `Route<StringsDict, UnknownDict, any>`, so this is sort of a shiv to make it work

Reply:
Unless there is something I am missing that would make us need to have separate API Wrappers for unvalidated and validated inputs, I think it would be better to stick to just a single "API Wrapper."  Looking to reduce complexity where possible.

Reply:
Yeah, we need this one because otherwise all the route types are "Unknown" and can't be used till typecasted. TypeScript won't compile until we convert them all :(